RELEASE.md

```diff
@@ -108,6 +109,16 @@ There may be new build versions of other dependencies (VS, MSVC).
 Our GitHub actions (tests) use the latest ones automatically,
 but the GitLab runner (release packages) doesn't.
 
+### Harden OpenSSL Defaults <a id="harden-ssl"></a>
+
+For major versions evaluate how far the global SSL defaults
+can be raised without violating the following requirements:
+
+* Icinga `2.x.y` must be able to communicate with version `2.{x-1}.0` by default
+* Icinga `2.x.y` should be able to communicate with version `2.{x-n}.0`
+  at least after overriding the defaults of version `2.x.y`,
+  the difficulty of this should be directly proportional to n
```

Comment:
Working with Icinga Web is also a requirement and other clients should also be considered (we provide a REST API for external users after all).

> the difficulty of this should be directly proportional to n

I'd state what you're trying to say in less technical terms, i.e. the older the version, the more acceptable it becomes that manual configuration is required. The way it's currently written, it reads a bit like we should actively make it more difficult for very old versions, even if we don't have to.

Reply:
Hmm... actively make it more difficult... now as you mention it, why not? 😈

Jokes aside: shouldn't we share our libssl with PHP on a system so that this isn't an issue?

Reply:
> Jokes aside: shouldn't we share our libssl with PHP on a system so that this isn't an issue?

In many cases probably yes, but there are more TLS implementations than OpenSSL. I don't what works with PHP, but at least libcurl can also use GnuTLS. Just wanted to point out that there are more components than Icinga 2 that have to stay compatible. Hopefully they don't need extra consideration in most cases.

Reply:
For TLS 1.3, the limiting factors currently are Linux distributions that still are on OpenSSL 1.0.2 (Amazon Linux 2, CentOS/RHEL 7, SLES 12). When all these are EOL, we can consider enforcing TLS 1.3+ (if there was a new standard by then) by default, but that's a discussion for in two years.